include/ck/utility/amd_inline_asm.hpp

```diff
@@ -5,7 +5,7 @@
 #define CK_AMD_INLINE_ASM_HPP
 
 #include "c_style_pointer_cast.hpp"
-#include "data_type.hpp"
+#include "dtype_vector.hpp"
```

Comment:
note, you actually don't need the header here if you forward-declare the vector types (and maybe replace the vector type function arguments with pointers)